### ISSUE TITLE
[QoL] [Newleaf] [Photographer] Choosing a card to copy

### ISSUE THREAD

**Issue by Dajakk** [tadeaustria/EverdellScore#35]
Right now popup for selecting a card is displayed only when adding Photographer to the town. This forces player to add cards in a specific order, especially if the opponent has a Photographer too (which forces not only an order, but also requires removal of a card and adding it back). In my opinion choice should be delayed until the towns are full.

[Idea] Current implementation is very nice, but it's also helping a lot, maybe too much for some players? Or too little for some lazy ones? Also it is a little bit misleading when the town is not full yet. I would consider adding an option to switch between:
- not showing how many points each card will give, player has to count and decide for himself what is the best for him
- current implementation
- automatic choice of the best card (like it works for example for Rug the Ruler and Flynn Nobletail)

[Idea] Implement similar setting for other cards that give players the choice of the opponent, like Rug the Ruler and Flynn Nobletail (any others I've missed?)

**Comment by Dajakk** [comment]
One more thing. When you add a Photographer to your town and choose a card to copy.... and then you remove the card that is copied (because for example it was added by mistake), it is still copied by Photographer, the only way to fix it is by removing her and adding again

**Comment by tadeaustria** [comment]
> One more thing. When you add a Photographer to your town and choose a card to copy.... and then you remove the card that is copied (because for example it was added by mistake), it is still copied by Photographer, the only way to fix it is by removing her and adding again

Yes, this is by design. Any other implementation would be too complicated.

**Comment by tadeaustria** [comment]
> [Idea] Current implementation is very nice, but it's also helping a lot, maybe too much for some players? Or too little for some lazy ones? Also it is a little bit misleading when the town is not full yet. I would consider adding an option to switch between:

Yes, I see your point there. I could think of one setting for all of such cards. 
Technically I think, a player will always choose his "best" option no matter how long it would take to think about, but I agree that players would like to think on their own, about the choice.

**Comment by tadeaustria** [comment]
> Right now popup for selecting a card is displayed only when adding Photographer to the town. This forces player to add cards in a specific order, especially if the opponent has a Photographer too (which forces not only an order, but also requires removal of a card and adding it back). In my opinion choice should be delayed until the towns are full.

Yes, currently, it is a manual step to choose the photographer at the end of the game. Another implementation could be, that any prosperity card can be chosen instead of checking against other players. Maybe an option to choose this at the end would be nice as well.

**Comment by Dajakk** [comment]
> > [Idea] Current implementation is very nice, but it's also helping a lot, maybe too much for some players? Or too little for some lazy ones? Also it is a little bit misleading when the town is not full yet. I would consider adding an option to switch between:
> 
> Yes, I see your point there. I could think of one setting for all of such cards. Technically I think, a player will always choose his "best" option no matter how long it would take to think about, but I agree that players would like to think on their own, about the choice.

Unless he makes a human error, which might be part of the fun?

**Comment by Dajakk** [comment]
> Yes, currently, it is a manual step to choose the photographer at the end of the game. Another implementation could be, that any prosperity card can be chosen instead of checking against other players. Maybe an option to choose this at the end would be nice as well.

I just had an idea. Maybe when adding Photographer he could be not paired with any other card, the score would show for example "+ ?" points, this could be reflected also on the player tab, so players would know immediately there is some action needed. When clicked, the "?" would open the current popup?
